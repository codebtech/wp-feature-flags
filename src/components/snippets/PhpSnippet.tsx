import Snippet from './Snippet';
import { useMemo } from '@wordpress/element';
import { __ } from '@wordpress/i18n';
import Clipboard from '../common/Clipboard';

export default function ({ flag }: { flag: string }): JSX.Element {
	const phpSnippet = useMemo(() => {
		return `if ( class_exists( '\\MR\\FeatureFlags\\Utils' ) && \\MR\\FeatureFlags\\Utils::is_enabled( '${flag}' ) ) {
	// php code goes here...
}`;
	}, [flag]);
	return (
		<div className="mr-feature-flag-php-snippet-container">
			<h3>{__('PHP Snippet', 'mr-feature-flags')}</h3>
			<Clipboard
				text={phpSnippet}
				style={{
					color: 'darkgray',
					float: 'right',
					position: 'relative',
					right: 40,
					top: 24,
				}}
			/>
			<Snippet data={phpSnippet} language={'php'} />
		</div>
	);
}
