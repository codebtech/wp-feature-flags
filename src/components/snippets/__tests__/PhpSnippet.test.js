import PhpSnippet from '../PhpSnippet';
import { render } from '@testing-library/react';

describe('PhpSnippet component', () => {
	it('it should render the correct PHP snippet', () => {
		const { getByText } = render(<PhpSnippet flag={'testFlag'} />);
		expect(getByText(/testFlag/i)).toBeInTheDocument();
	});

	it('matches snapshot', () => {
		const { asFragment } = render(<PhpSnippet flag={'testFlag'} />);
		expect(asFragment()).toMatchSnapshot();
	});
});
