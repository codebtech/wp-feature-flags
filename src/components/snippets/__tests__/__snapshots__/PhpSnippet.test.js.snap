// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PhpSnippet component matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="mr-feature-flag-php-snippet-container"
  >
    <h3>
      PHP Snippet
    </h3>
    <button
      aria-label="Copy to clipboard"
      aria-pressed="false"
      class="components-button codeb-feature-flags-clipboard-base is-tertiary has-icon"
      type="button"
    >
      <span
        class="dashicon dashicons dashicons-clipboard"
      />
    </button>
    <pre
      style="display: block; overflow-x: auto; background: rgb(43, 43, 43); color: rgb(248, 248, 242); padding: 0.5em;"
    >
      <code
        class="language-php"
        style="white-space: pre;"
      >
        <span
          style="color: rgb(220, 198, 224);"
        >
          use
        </span>
        <span>
           
        </span>
        <span
          style="color: rgb(0, 224, 224);"
        >
          CodeB
        </span>
        <span>
          \\
        </span>
        <span
          style="color: rgb(0, 224, 224);"
        >
          FeatureFlags
        </span>
        <span>
          \\
        </span>
        <span
          style="color: rgb(0, 224, 224);"
        >
          Flag
        </span>
        <span>
          ;

        </span>
        <span />
        <span
          style="color: rgb(220, 198, 224);"
        >
          if
        </span>
        <span>
           ( class_exists( 
        </span>
        <span
          style="color: rgb(171, 227, 56);"
        >
          '\\CodeB\\FeatureFlags\\Flag'
        </span>
        <span>
           ) && Flag::is_enabled( 
        </span>
        <span
          style="color: rgb(171, 227, 56);"
        >
          'testFlag'
        </span>
        <span>
           ) ) {

        </span>
        <span>
          	
        </span>
        <span
          style="color: rgb(212, 208, 171);"
        >
          // php code goes here...
        </span>
        <span>
          

        </span>
        }
      </code>
    </pre>
  </div>
</DocumentFragment>
`;
